src/ADAL.PCL/TokenCacheKey.cs

```diff
@@ -156,6 +171,22 @@ public bool ScopeIntersects(string[] otherScope)
             return this.Scope.Intersect(otherScope).ToArray().Length > 0;
         }
 
+        internal bool PolicyEquals(string otherPolicy)
+        {
+            if (string.IsNullOrEmpty(this.Policy) && string.IsNullOrEmpty(otherPolicy))
+            {
+                return true;
+            }
+
+            if (string.IsNullOrEmpty(this.Policy) || string.IsNullOrEmpty(otherPolicy))
+            {
+                return false;
+            }
+
+
+            return (string.Compare(otherPolicy, this.Policy, StringComparison.OrdinalIgnoreCase) == 0);
```

Comment:
only string.compare should do. https://msdn.microsoft.com/en-us/library/84787k22(v=vs.110).aspx